laser/photon_jit/photon_osalloc.nim

```diff
@@ -54,26 +61,55 @@ func flag*[E: enum](e: varargs[E]): Flag[E] {.inline.} =
   result = Flag[E](flags)
 
 when not defined(release):
-  block:
-    var DebugMapPrivate {.importc: "MAP_PRIVATE", header: "<sys/mman.h>".}: cint
-    var DebugMapAnonymous {.importc: "MAP_ANONYMOUS", header: "<sys/mman.h>".}: cint
+  when defined(windows):
+      var DebugMemCommit {.importc: "MEM_COMMIT", header: "<windows.h>".}: cint
+      var DebugMemRelease {.importc: "MEM_RELEASE", header: "<windows.h>".}: cint
+
+      assert ord(MemCommit) == DebugMemCommit, "Your CPU+OS platform is misconfigured"
+      assert ord(MemRelease) == DebugMemRelease, "Your CPU+OS platform is misconfigured"
+  else:
+    block:
+      var DebugMapPrivate {.importc: "MAP_PRIVATE", header: "<sys/mman.h>".}: cint
+      var DebugMapAnonymous {.importc: "MAP_ANONYMOUS", header: "<sys/mman.h>".}: cint
+
+      assert ord(MapPrivate) == DebugMapPrivate, "Your CPU+OS platform is misconfigured"
+      assert ord(MapAnonymous) == DebugMapAnonymous, "Your CPU+OS platform is misconfigured"
+
+when defined(windows):
+  proc VirtualAlloc*(
+      lpAddress: pointer,
+      dwSize: int;
+      flAllocationType: Flag[MemAlloc],
+      flProtect: MemProt
+    ): pointer {.importc, stdcall, dynlib: "Kernel32", sideeffect.}
```

Comment:
Not on windows at the moment but I think I need the header:"<windows.h>"

Reply:
you can grab VirtualAlloc directly from the DLL, importing `windows.h` is  not necessary. https://github.com/nim-lang/oldwinapi/blob/master/oldwinapi/windows.nim#L18570

Reply:
actually I thought about it for a minute and came to the conclusion that if Nim ever supports C++/WinRT it's better to use the header since you can't link to DLLs (you have to use "`VirtualProtectFromApp()`" instead of "`VirtualProtect()`")